mcs/class/System.Core/Test/System.IO.MemoryMappedFiles/MemoryMappedFileTest.cs

```diff
@@ -77,9 +89,14 @@ static String MkNamedMapping ()
 		}
 
 		[TearDown]
-		protected void TearDown () {
-			if (Directory.Exists (tempDir))
+		public void TearDown ()
+		{
+			try {
+				// This throws an exception under MS.NET and Mono on Windows,
+				// since the directory contains open files.
 				Directory.Delete (tempDir, true);
+			} catch (Exception) {
```

Comment:
Yeah, making sure everything is closed at the end of every test would be ideal. But it's hard to make sure that's the case when a test fails for some reason. And to cope with that we need to make sure every test uses new temp files otherwise the following tests will fail on Windows due to the file already being open and thus being locked from deletion. We do this in other tests already (e.g. AssemblyTest) to work around the problem with locked files on Windows.